pyannote/pipeline/pipeline.py

```diff
@@ -394,6 +395,13 @@ def instantiate(self, params: dict) -> 'Pipeline':
 
             # instantiate parameter value
             if name in self._parameters:
+                # skip Frozen parameters
+                param = getattr(self, name)
+                if isinstance(param, Frozen) and param.value != value:
+                    msg = (f"using frozen value {param.value:g} instead of provided " 
+                           f"value {value:g} for parameter {name} ")
+                    warn(msg)
```

Comment:
Actually, shouldn't we add somthing like `setattr(self, name, param.value)` here ?
I think I misunderstood the usage of `Frozen` parameters.

When training a pipeline I got a `TypeError`:
```
[W 2020-05-25 12:22:21,110] Setting status of trial#0 as TrialState.FAIL because of the following error: TypeError("'>' not supported between instances of 'float' and 'Frozen'")
Traceback (most recent call last):
  File "/mnt/beegfs/home/lerner/anaconda3/envs/pyannote/lib/python3.7/site-packages/optuna/study.py", line 648, in _run_trial
    result = func(trial)
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-pipeline/pyannote/pipeline/optimizer.py", line 184, in objective
    output = pipeline(input)
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-audio/pyannote/audio/pipeline/speaker_diarization.py", line 154, in __call__
    speech_turns = self.speech_turn_segmentation(current_file)
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-audio/pyannote/audio/pipeline/speech_turn_segmentation.py", line 123, in __call__
    sad = self.speech_activity_detection(current_file).get_timeline()
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-audio/pyannote/audio/pipeline/speech_activity_detection.py", line 159, in __call__
    speech = self._binarize.apply(speech_prob)
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-audio/pyannote/audio/utils/signal.py", line 201, in apply
    label = data[0] > self.onset
TypeError: '>' not supported between instances of 'float' and 'Frozen'
Traceback (most recent call last):
  File "/mnt/beegfs/home/lerner/anaconda3/envs/pyannote/bin/pyannote-pipeline", line 11, in <module>
    load_entry_point('pyannote.pipeline', 'console_scripts', 'pyannote-pipeline')()
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-pipeline/pyannote/pipeline/experiment.py", line 427, in main
    pretrained=pretrained, sampler=sampler, pruner=pruner)
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-pipeline/pyannote/pipeline/experiment.py", line 281, in train
    for i, status in zip(count, iterations):
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-pipeline/pyannote/pipeline/optimizer.py", line 288, in tune_iter
    timeout=None, n_jobs=1)
  File "/mnt/beegfs/home/lerner/anaconda3/envs/pyannote/lib/python3.7/site-packages/optuna/study.py", line 319, in optimize
    gc_after_trial, None)
  File "/mnt/beegfs/home/lerner/anaconda3/envs/pyannote/lib/python3.7/site-packages/optuna/study.py", line 597, in _optimize_sequential
    self._run_trial_and_callbacks(func, catch, callbacks, gc_after_trial)
  File "/mnt/beegfs/home/lerner/anaconda3/envs/pyannote/lib/python3.7/site-packages/optuna/study.py", line 627, in _run_trial_and_callbacks
    trial = self._run_trial(func, catch, gc_after_trial)
  File "/mnt/beegfs/home/lerner/anaconda3/envs/pyannote/lib/python3.7/site-packages/optuna/study.py", line 648, in _run_trial
    result = func(trial)
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-pipeline/pyannote/pipeline/optimizer.py", line 184, in objective
    output = pipeline(input)
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-audio/pyannote/audio/pipeline/speaker_diarization.py", line 154, in __call__
    speech_turns = self.speech_turn_segmentation(current_file)
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-audio/pyannote/audio/pipeline/speech_turn_segmentation.py", line 123, in __call__
    sad = self.speech_activity_detection(current_file).get_timeline()
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-audio/pyannote/audio/pipeline/speech_activity_detection.py", line 159, in __call__
    speech = self._binarize.apply(speech_prob)
  File "/mnt/beegfs/projects/plumcot/pyannote/pyannote-audio/pyannote/audio/utils/signal.py", line 201, in apply
    label = data[0] > self.onset
TypeError: '>' not supported between instances of 'float' and 'Frozen'
Waiting for first iteration to complete: : 0iteration [00:12, ?iteration/s]
```

Reply:
Oh, yes. Better: set `value = param.value` in the `if` statement and remove the `continue`.

Reply:
We could also fix issue #4 in this pull request if we do that differently.
The idea would be to loop on self.parameters(frozen=True) and add (issue #4) or update (this issue) them in provided `params`.